lib/zlib.toit

```diff
@@ -8,99 +8,86 @@ import crypto
 import crypto.adler32 as crypto
 import crypto.crc as crc-algorithms
 
-class RleReader implements reader.Reader:
-  owner_ /ZlibEncoder_? := null
+class CompressionReader implements reader.Reader:
+  owner_ := null
 
   constructor.private_:
 
-  read:
-    return owner_.read_
+  read --wait/bool=true -> ByteArray?:
+    return owner_.read_ --wait=wait
 
   close:
-    // Currently does nothing.
-
-// TODO: Should not be a Reader, but we have deprecated the read method,
-// and there's no way to deprecate an 'implements' statement. This class
-// is a superclass for some public classes, so we can't just remove it
-// in a backwards compatible way.
-class ZlibEncoder_ implements reader.Reader:
-  channel_ := monitor.Channel 1
-  reader/RleReader ::= RleReader.private_
-
-  static SMALL-BUFFER-DEFLATE-HEADER_ ::= [8, 0x1d]
-  static MINIMAL-GZIP-HEADER_ ::= [0x1f, 0x8b, 8, 0, 0, 0, 0, 0, 0, 0xff]
-
-  constructor --gzip-header/bool:
-    reader.owner_ = this
-    header := gzip-header ? MINIMAL-GZIP-HEADER_ : SMALL-BUFFER-DEFLATE-HEADER_
-    channel_.send (ByteArray header.size: header[it])
-
-  /**
-  Deprecated.
-  Use the $reader to get an object with a read method.
-  */
-  read:
-    return read_
+    owner_.close-read_
 
-  read_:
-    result := channel_.receive
-    if not result: channel_.send null  // Once it's closed we should be able to keep reading nulls.
-    return result
+SMALL-BUFFER-DEFLATE-HEADER_ ::= [8, 0x1d]
+MINIMAL-GZIP-HEADER_ ::= [0x1f, 0x8b, 8, 0, 0, 0, 0, 0, 0, 0xff]
 
-class UncompressedDeflateEncoder_ extends ZlibEncoder_:
-  summer_/crypto.Checksum? := ?
+class UncompressedDeflateBackEnd_ implements BackEnd_:
```

Comment:
Add a bit of documentation, to help the reader.
Basically, pipes the data through in 256byte blocks. Needs to maintain a checksum...